doc: generate v1.13.0 (#1639)
mongodb/terraform-provider-mongodbatlas: CHANGELOG.md

# Changelog

> ## [v1.13.0](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/v1.13.0) (2023-11-17)

Reviewer: ```suggestion
## [v1.13.0](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/v1.13.0) (2023-11-21)
```

[Full Changelog](https://github.com/mongodb/terraform-provider-mongodbatlas/compare/v1.12.3...v1.13.0)

**Enhancements**

> - Adds support for search node management with `mongodbatlas_search_deployment` resource and data source [\#1633](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1633) ([AgustinBettati](https://github.com/AgustinBettati))

Reviewer: ```suggestion
- Adds support for [MongoDB Atlas Search Node](https://www.mongodb.com/docs/atlas/atlas-search/atlas-search-overview/#search-nodes-architecture) management with `mongodbatlas_search_deployment` resource and data source [\#1633](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1633) ([AgustinBettati](https://github.com/AgustinBettati))
```

> - Adds type and fields attributes in resource and data sources for `search_index` [\#1605](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1605) ([lantoli](https://github.com/lantoli))

Reviewer 2: ```suggestion
- Adds `type` and `fields` attributes in resource and data sources for `search_index` [\#1605](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1605) ([lantoli](https://github.com/lantoli))
```

> - Adds type and fields attributes in resource and data sources for `search_index` [\#1605](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1605) ([lantoli](https://github.com/lantoli))

Reviewer: ```suggestion
- Adds 'vectorSearch` type and fields attributes in resource and data sources for `mongodbatlas_search_index` [\#1605](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1605) ([lantoli](https://github.com/lantoli))
```

**Bug Fixes**

> - Fixes terraform encryption at rest error when upgrading to 1.12.2 [\#1617](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1617) ([andreaangiolillo](https://github.com/andreaangiolillo))

Reviewer 2: ```suggestion
- Fixes Terraform encryption-at-rest error when upgrading to 1.12.2 [\#1617](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1617) ([andreaangiolillo](https://github.com/andreaangiolillo))
```

"encryption at rest" is being used to describe "error", the hyphen helps make it clear the entire phrase is used for that purpose.

> - Fixes terraform encryption at rest error when upgrading to 1.12.2 [\#1617](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1617) ([andreaangiolillo](https://github.com/andreaangiolillo))

Reviewer: ```suggestion
- Fixes terraform encryption at rest error when upgrading to Terraform Provider for MongoDB Atlas v1.12.2 [\#1617](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1617) ([andreaangiolillo](https://github.com/andreaangiolillo))
```

> - Defines `ConfigMode` for computed nest block attribute in `privatelink_endpoint_service` to support tf 1.0.8 [\#1629](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1629) ([AgustinBettati](https://github.com/AgustinBettati))

Reviewer 2: ```suggestion
- Defines `ConfigMode` for the `endpoints`  attribute, enabling use of computed nested blocks in`privatelink_endpoint_service` to support tf 1.0.8 [\#1629](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1629) ([AgustinBettati](https://github.com/AgustinBettati))
```

> - Defines `ConfigMode` for computed nest block attribute in `privatelink_endpoint_service` to support tf 1.0.8 [\#1629](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1629) ([AgustinBettati](https://github.com/AgustinBettati))

Reviewer: ```suggestion
- Defines `ConfigMode` for computed nest block attribute in `mongodbatlas_privatelink_endpoint_service` to support tf 1.0.8 [\#1629](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1629) ([AgustinBettati](https://github.com/AgustinBettati))
```

> - Defines `ConfigMode` for computed nest block attribute in `privatelink_endpoint_service` to support tf 1.0.8 [\#1629](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1629) ([AgustinBettati](https://github.com/AgustinBettati))

Reviewer: ```suggestion
- Defines `ConfigMode` for computed nest block attribute in `privatelink_endpoint_service` to support HashiCorp Terraform version 1.0.8 [\#1629](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1629) ([AgustinBettati](https://github.com/AgustinBettati))
```

**Closed Issues**

- `tags` not working for cluster [\#1619](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1619)

> - Terraform Bud in updating serverless project [\#1611](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1611)

Reviewer 2: ```suggestion
- Terraform Bug in updating serverless project [\#1611](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1611)
```

- Changes to `mongodbatlas_project_ip_access_list` comments force a replacement [\#1600](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1600)
- Rate limit of 10 invitations per 1 minutes exceeded [\#1589](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1589)

> - The terraform-provider-mongodbatlas_v1.12.2 plugin crashed! [\#1567](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1567)

Reviewer 2: ```suggestion
- The `terraform-provider-mongodbatlas_v1.12.2` plugin crashed! [\#1567](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1567)
```

- Breaking change to drop deprecated fields made in minor version release [\#1493](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1493)

**Internal Improvements**

- Updates PR action to automatically add labels based on the PR title [\#1637](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1637) ([andreaangiolillo](https://github.com/andreaangiolillo))
- Removes unused code in resource and cleanup in `project_api_key` docs [\#1636](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1636) ([AgustinBettati](https://github.com/AgustinBettati))

> - Improves testing in search-index resource [\#1635](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1635) ([lantoli](https://github.com/lantoli))

Reviewer 2: ```suggestion
- Improves testing of `search_index` resource [\#1635](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1635) ([lantoli](https://github.com/lantoli))
```

- Updates Atlas Go SDK [\#1632](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1632) ([lantoli](https://github.com/lantoli))
- Updates PR template for further verifications. [\#1628](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1628) ([marcosuma](https://github.com/marcosuma))
- Makes `disk_iops` a computed attribute in `advanced_cluster` resource [\#1620](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1620) ([AgustinBettati](https://github.com/AgustinBettati))

> - Removes all references to flowdock and new relic third party integrations [\#1616](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1616) ([oarbusi](https://github.com/oarbusi))

Reviewer 2: ```suggestion
- Removes all references to Flowdock and New Relic third-party integrations [\#1616](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1616) ([oarbusi](https://github.com/oarbusi))
```

- Fixes documentation errors in cloud provider access [\#1615](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1615) ([milh0use](https://github.com/milh0use))

> - Fixes search index acceptance tests flow [\#1610](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1610) ([lantoli](https://github.com/lantoli))

Reviewer 2: ```suggestion
- Fixes `search_index` acceptance tests flow [\#1610](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1610) ([lantoli](https://github.com/lantoli))
```

# Changelog

## [v1.13.0](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/v1.13.0) (2023-11-17)

[Full Changelog](https://github.com/mongodb/terraform-provider-mongodbatlas/compare/v1.12.3...v1.13.0)

**Enhancements**

> - feat: search node management with `mongodbatlas_search_deployment` resource and data source [\#1633](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1633) ([AgustinBettati](https://github.com/AgustinBettati))

Reviewer 3: Adds support for search node...

**Bug Fixes**

> - fix: Terraform encryption at rest error when upgrading to 1.12.2 [\#1617](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1617) ([andreaangiolillo](https://github.com/andreaangiolillo))

Reviewer 3: Fixes Terraform encryption...

**Closed Issues**

> - 'tags' not working for cluster [\#1619](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1619)

Reviewer 3: tags with ticks ` instead of single quotes

- Rate limit of 10 invitations per 1 minutes exceeded [\#1589](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1589)
- The terraform-provider-mongodbatlas_v1.12.2 plugin crashed! [\#1567](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1567)
- Breaking change to drop deprecated fields made in minor version release [\#1493](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1493)

**Internal Improvements**

- Updates PR action to automatically add labels based on the PR title [\#1637](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1637) ([andreaangiolillo](https://github.com/andreaangiolillo))
- Removes unused code in resource and cleanup in `project_api_key` docs [\#1636](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1636) ([AgustinBettati](https://github.com/AgustinBettati))
- Improves testing in search-index resource [\#1635](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1635) ([lantoli](https://github.com/lantoli))
- Updates Atlas Go SDK [\#1632](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1632) ([lantoli](https://github.com/lantoli))

> - Defines `ConfigMode` for computed nest block attribute in `privatelink_endpoint_service` to support tf 1.0.8 [\#1629](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1629) ([AgustinBettati](https://github.com/AgustinBettati))

Reviewer 4: This can go in bug fix section

- Bumps tj-actions/verify-changed-files from 6d688963a73d28584e163b6f62cf927a282c4d11 to 82a523f60ad6816c35b418520f84629024d70e1e [\#1626](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1626) ([dependabot[bot]](https://github.com/apps/dependabot))
- Bumps github.com/aws/aws-sdk-go from 1.47.5 to 1.47.11 [\#1625](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1625) ([dependabot[bot]](https://github.com/apps/dependabot))
- Bumps github.com/hashicorp/terraform-plugin-sdk/v2 from 2.29.0 to 2.30.0 [\#1624](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1624) ([dependabot[bot]](https://github.com/apps/dependabot))
- Bumps github.com/gruntwork-io/terratest from 0.46.6 to 0.46.7 [\#1623](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1623) ([dependabot[bot]](https://github.com/apps/dependabot))
- Makes `disk_iops` a computed attribute in `advanced_cluster` resource [\#1620](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1620) ([AgustinBettati](https://github.com/AgustinBettati))

> - Removes all references to flowdock and new relic third party integrations [\#1616](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1616) ([oarbusi](https://github.com/oarbusi))

Reviewer 3: I recommend all bumps together at the end of the section

Author: Grouped them all at the end of the section

- Fixes documentation errors in cloud provider access [\#1615](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1615) ([milh0use](https://github.com/milh0use))
- Fixes search index acceptance tests flow [\#1610](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1610) ([lantoli](https://github.com/lantoli))
- Bumps github.com/aws/aws-sdk-go from 1.47.4 to 1.47.5 [\#1608](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1608) ([dependabot[bot]](https://github.com/apps/dependabot))
- Bumps github.com/gruntwork-io/terratest from 0.46.1 to 0.46.6 [\#1607](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1607) ([dependabot[bot]](https://github.com/apps/dependabot))
- Bumps github.com/aws/aws-sdk-go from 1.46.3 to 1.47.4 [\#1606](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1606) ([dependabot[bot]](https://github.com/apps/dependabot))

> - Adds type and fields attributes in resource and data sources for `search_index` [\#1605](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1605) ([lantoli](https://github.com/lantoli))

Reviewer 4: the new vector search index feature should be with the enhancements section

Reviewer 5: +1

- Reduces stale days to 5 and close after 2 day of stale [\#1602](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1602) ([andreaangiolillo](https://github.com/andreaangiolillo))
- Fixes Update SDK GitHub action [\#1596](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1596) ([lantoli](https://github.com/lantoli))
- Disables `data_source` `event_trigger` tests since they are failing [\#1595](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1595) ([marcosuma](https://github.com/marcosuma))
- Changes naming convention for `data_source` event trigger test. [\#1594](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1594) ([marcosuma](https://github.com/marcosuma))
- Changes naming convention for `data_source` event trigger tests. [\#1593](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1593) ([marcosuma](https://github.com/marcosuma))
- Updates RELEASING.md [\#1592](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1592) ([andreaangiolillo](https://github.com/andreaangiolillo))

> - Reverts "chore: add acceptance tests and migration tests to report in merge queue" [\#1591](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1591) ([andreaangiolillo](https://github.com/andreaangiolillo))

Author: Should we include reverts in the CHANGELOG or should we remove this line and the next since they cancel each other?

Reviewer 4: I am in favour of removing

Reviewer 5: Since we noticed it we can remote, but overall I don't think it's that critical

Reviewer 3: if both the original and remote are in this release, they annihilate each other and we can just remove them both :-)

Author: Removed
